xcat-inventory/xcclient/inventory/backend.py

```diff
@@ -0,0 +1,696 @@
+from exceptions import * 
+import manager 
+import utils
+import pickle
+import hashlib
+
+class Invbackend:
+
+    globalcfgpath="/etc/xcat/inventory.cfg"
+    bkendcfg={}
+
+    def loadcfg(self,cfgpath=None):
+        if cfgpath is None:
+            myhome=utils.gethome()
+            cfgpath=myhome+'/.xcatinv/inventory.cfg' 
+        if not os.path.isfile(cfgpath):
+            cfgpath=self.globalcfgpath 
+
+        if not os.path.exists(cfgpath):
+            raise InvalidFileException("The configuration file %s does not exist!\n"%(cfgpath))    
+
+        config = configparser.ConfigParser()
+        config.read(cfgpath)
+        if not config:
+            raise ParseException("Unable to parse configuration file %s"%(cfgpath))
+       
+        if 'type' in config['backend'].keys():
+            self.bkendcfg['type']=config['backend']['type'].strip('"').strip("'")
+        else:
+            self.bkendcfg['type']='git'
+
+        if 'workspace' in config['backend'].keys():
+            self.bkendcfg['workspace']=config['backend']['workspace'].strip('"').strip("'")
+        else:
+            self.bkendcfg['workspace']=''
+
+        if 'user' in config['backend'].keys():
+            self.bkendcfg['user']=config['backend']['user'].strip('"').strip("'")
+        else:
+            self.bkendcfg['user']="xcat"
+
+        if 'email' in config['backend'].keys():
+            self.bkendcfg['email']=config['backend']['email'].strip('"').strip("'")
+        else: 
+            self.bkendcfg['email']="xcat@xcat.org"
+
+        self.bkendcfg['InfraRepo']={}
+        if 'InfraRepo' in config.keys():
+            if 'remote_repo' in config['InfraRepo'].keys(): 
+                self.bkendcfg['InfraRepo']['remote_repo']=config['InfraRepo']['remote_repo'].strip('"').strip("'")
+            else:
+                self.bkendcfg['InfraRepo']['remote_repo']=""
+
+            if 'local_repo' in config['InfraRepo'].keys(): 
+                self.bkendcfg['InfraRepo']['local_repo']=config['InfraRepo']['local_repo'].strip('"').strip("'")
+            else:
+                self.bkendcfg['InfraRepo']['local_repo']=utils.gethome()
+
+            if 'working_dir' in config['InfraRepo'].keys(): 
+                self.bkendcfg['InfraRepo']['working_dir']=config['InfraRepo']['working_dir'].strip('"').strip("'")
+            else:
+                self.bkendcfg['InfraRepo']['working_dir']='.'
+             
+
+
+    def __initcfgfile(self):
+        myhome=utils.gethome()
+        cfgpath=myhome+'/.xcatinv/inventory.cfg'
+        if not os.path.exists(cfgpath):
+            if not os.path.exists(self.globalcfgpath):
+                raise FileNotExistException('File "%s" does not exist, please check ...' % self.globalcfgpath) 
+            os.makedirs(myhome+'/.xcatinv')
+            shutil.copyfile(self.globalcfgpath, cfgpath)   
+            print("local configuration file \"%s\" created, you can customize it"%(cfgpath))
+
+    def init(self,cfgpath=None): 
```

Comment:
Lots of the PEP8 coding style issue, just like need space for the operater.  Before commit, we can load it into PyCharm, and it will help your format it automatically with better PEP8 compliance.